sidekick_core/lib/src/sidekick_context.dart

```diff
@@ -0,0 +1,206 @@
+  // ///
+  // /// It's optional, not every project has a mainProject, there are repositories
+  // /// with zero or multiple projects.
+  // static DartPackage? get mainProject => throw 'TODO';
+
+  /// The location of the sidekick package inside the [repository]
+  static Directory get sidekickPackageDir => sidekickPackage.root;
+
+  /// The sidekick package inside the [repository]
+  static SidekickPackage get sidekickPackage {
+    return _cache.getOrCreate('sidekickPackage', _findSidekickPackage);
+  }
+
+  static SidekickPackage _findSidekickPackage() {
+    final injectedPackageHome = env[_envPackageHome];
+    if (injectedPackageHome != null && injectedPackageHome.isNotBlank) {
+      // When called via entryPoint, immediately return the package
+      return SidekickPackage.fromDirectory(Directory(injectedPackageHome))!;
+    }
+
+    final script = File(DartScript.self.pathToScript);
+    final scriptPath = script.uri.path;
+    final pathSep = Platform.pathSeparator;
+
+    // When CLI is run with compiled entryPoint: <sidekick package>/build/cli.exe
+    if (scriptPath.endsWith('${pathSep}build${pathSep}cli.exe')) {
+      return SidekickPackage.fromDirectory(script.parent.parent)!;
+    }
+
+    // When CLI is run with `dart bin/main.dart`: <sidekick package>/bin/main.dart
+    if (scriptPath.endsWith('${pathSep}bin${pathSep}main.dart')) {
+      return SidekickPackage.fromDirectory(script.parent.parent)!;
+    }
+
+    // in `UpdateCommand` when the latest `update_sidekick_cli.dart` is written to <sidekick package>/build/update.dart to be executed
+    if (scriptPath.endsWith('${pathSep}bin${pathSep}update.dart')) {
+      return SidekickPackage.fromDirectory(script.parent.parent)!;
+    }
+
+    // Fallback strategy: searching all parent folders until we find a dart package root
+    final scriptDirectory = script.parent;
+    Directory current = Directory(scriptDirectory.path);
+    final repoRoot = repository.root;
+
+    while (current.isWithin(repoRoot)) {
+      final sidekickPackage = SidekickPackage.fromDirectory(current);
+      if (sidekickPackage != null) {
+        return sidekickPackage;
+      }
+
+      current = current.parent;
+    }
+    throw "Can't find sidekickPackage from ${scriptDirectory.path}";
+  }
+
+  /// The location of the entryPoint inside the [repository]
+  ///
+  /// Usually injected from the entryPoint itself via `env.SIDEKICK_ENTRYPOINT_HOME`
+  static EntryPoint get entryPoint {
+    return _cache.getOrCreate('findEntryPoint', _findEntryPoint);
+  }
+
+  static EntryPoint _findEntryPoint() {
+    if (env.exists(_envEntryPointHome)) {
+      // CLI is called via entryPoint
+      final injectedEntryPointPath = env[_envEntryPointHome];
+      if (injectedEntryPointPath == null || injectedEntryPointPath.isBlank) {
+        throw 'Injected entryPoint was not set (env.$_envEntryPointHome)';
+      }
+      final entryPointFile =
+          File(normalize('$injectedEntryPointPath/${core.cliName}'));
```

Comment:
We should not use `core.cliName` which is set in `initializeSidekick`. It is now always available.

Instead we should change the `entryPoint` shell script to expose `SIDEKICK_ENTRYPOINT_PATH`, instead of `SIDEKICK_ENTRYPOINT_HOME`
We already have this information and only need to expose it

```bash

export SIDEKICK_ENTRYPOINT_PATH="${PRG}"
```